src/sentry/identity/vsts/provider.py

```diff
@@ -69,12 +76,25 @@ def get_refresh_token_headers(self):
         return {"Content-Type": "application/x-www-form-urlencoded", "Content-Length": "1654"}
 
     def get_refresh_token_params(self, refresh_token, *args, **kwargs):
+        identity = kwargs.get("identity")
+        client_secret = options.get("vsts.client-secret")
+
+        # The token refresh flow does not operate within a pipeline in the same way
+        # that installation does, this means that we have to use the identity.scopes
+        # to determine which client_secret to use.
+        #
+        # If "vso.code" is missing from the identity.scopes, we know that we installed
+        # using the "vsts-limited.client-secret" and therefore should use that to refresh
+        # the token.
+        if "vso.code" not in identity.scopes:
+            client_secret = options.get("vsts-limited.client-secret")
```

Comment:
When we're in limited permissions mode are there any changes that need to be made for commit webhooks or release creation? Those seem like places that might need read access to code in azure.

Reply:
My understanding is that the repository functionality will not be able to be utilized at all, which is understood by the customer. So basically just using Azure for product management.

cc @manuzope